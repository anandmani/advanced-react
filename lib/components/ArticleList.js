import React from 'react'
import Article from './Article'

const ArticleList = ({ articles }) => {
  return <div>
    {
      Object.values(articles).map((article) => <Article
        key={article.id}
        article={article}
      />
      )
    }
  </div>
}

export default ArticleList