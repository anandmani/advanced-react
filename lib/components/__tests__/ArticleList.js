import React from 'react'
// import renderer from 'react-test-renderer'
import ArticleList from '../ArticleList'
import { shallow } from 'enzyme'

//TODO: add test for context

require('babel-polyfill') //https://github.com/facebook/jest/issues/3687    Jest doesn't require babel-polyfill so, Object.values is not supported in jest.
// Object.values = (obj) => Object.keys(obj).map((key) => obj[key])

describe('ArticleList', () => {

  it('renders correctly', () => {
    const testProps = {
      articles: {
        1: {
          id: '1',
          title: 'title',
          date: 'date',
          body: 'body'
        },
        2: {
          id: '2',
          title: 'title',
          date: 'date',
          body: 'body'
        }
      },
      // store: {  //Removing store as a prop. Moving it to context
      //   getAuthor: () => ({})
      //   // getAuthor: jest.fn(() => ({}))  //when to use function mocks?
      // }
    }

    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    )

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('ArticleContainer')).toHaveLength(2)
  })

})