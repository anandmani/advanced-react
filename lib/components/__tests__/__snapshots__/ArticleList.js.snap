// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <ArticleContainer
      article={
        Object {
          "body": "body",
          "date": "date",
          "id": "1",
          "title": "title",
        }
      }
    />
    <ArticleContainer
      article={
        Object {
          "body": "body",
          "date": "date",
          "id": "2",
          "title": "title",
        }
      }
    />
  </div>,
  "nodes": Array [
    <div>
      <ArticleContainer
        article={
          Object {
            "body": "body",
            "date": "date",
            "id": "1",
            "title": "title",
          }
        }
      />
      <ArticleContainer
        article={
          Object {
            "body": "body",
            "date": "date",
            "id": "2",
            "title": "title",
          }
        }
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ArticleList
        articles={
          Object {
            "1": Object {
              "body": "body",
              "date": "date",
              "id": "1",
              "title": "title",
            },
            "2": Object {
              "body": "body",
              "date": "date",
              "id": "2",
              "title": "title",
            },
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "articles": Object {
            "1": Object {
              "body": "body",
              "date": "date",
              "id": "1",
              "title": "title",
            },
            "2": Object {
              "body": "body",
              "date": "date",
              "id": "2",
              "title": "title",
            },
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <ArticleContainer
            article={
              Object {
                "body": "body",
                "date": "date",
                "id": "1",
                "title": "title",
              }
            }
          />
          <ArticleContainer
            article={
              Object {
                "body": "body",
                "date": "date",
                "id": "2",
                "title": "title",
              }
            }
          />
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <ArticleContainer
            article={
              Object {
                "body": "body",
                "date": "date",
                "id": "1",
                "title": "title",
              }
            }
          />
          <ArticleContainer
            article={
              Object {
                "body": "body",
                "date": "date",
                "id": "2",
                "title": "title",
              }
            }
          />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ArticleList
    articles={
      Object {
        "1": Object {
          "body": "body",
          "date": "date",
          "id": "1",
          "title": "title",
        },
        "2": Object {
          "body": "body",
          "date": "date",
          "id": "2",
          "title": "title",
        },
      }
    }
  />,
}
`;
